Plugin.cs

```diff
@@ -158,6 +192,7 @@ private void OnNetworkEvent(IntPtr dataPtr, ushort opCode, uint sourceActorId, u
                 while (i < listing.ItemListings.Count && !listing.ItemListings[i].IsHq) i++;
                 if (i == listing.ItemListings.Count) return;
             }
+            while(i != listing.ItemListings.Count && this.configuration.whitelist.Contains(listing.ItemListings[i].RetainerName)) i++;
```

Comment:
This isn't quite correct. You have a chance of passing `i = listing.ItemListings.Count` to the next line, resulting in an out of bounds array access. This also interacts poorly with HQ mode.